docs/sources/configure-client/grafana-agent/go_pull.md

```diff
@@ -1,258 +1,224 @@
+                    target_label  = "container"
+            }
+    
+            rule {
+                    action        = "replace"
+                    regex         = "(.*)@(.*)"
+                    replacement   = "${1}/${2}"
+                    separator     = "@"
+                    source_labels = ["__meta_kubernetes_namespace", "__meta_kubernetes_pod_container_name"]
+                    target_label  = "service_name"
+            }
+            
+            rule { 
+                    action        = "keep"
+                    regex         = "(ns1/.*)|(ns2/container-.*0)"
+                    source_labels = ["service_name"]
+            }
+    }
+    ```
 
-To start using godeltaprof in pull mode in a Go application, you need to include godeltaprof module in your app:
+3. And then use `discovery.relabel.specific_pods.targets` as a target for `pyroscope.scrape` block.
 
-```bash
-go get github.com/grafana/pyroscope-go/godeltaprof@latest
-```
+    ```river
+        pyroscope.scrape "scrape_job_name" {
+                targets    = discovery.relabel.specific_pods.output
+                ...
+        }
+    ```
 
-and add it to your imports:
+### Exposing pprof endpoints
 
+If you don't use `http.DefaultServeMux` you can register `/debug/pprof/*` handlers to your own `http.ServeMux`
 ```go
-import _ "net/http/pprof"
-import _ "github.com/grafana/pyroscope-go/godeltaprof/http/pprof"// add this line as well
+var mux *http.ServeMux
+mux.Handle("/debug/pprof/", http.DefaultServeMux)
 ```
-
-If you do not have ability to update your code then disable all the `goddeltaprof_X` profiles and enable the corresponding standard `X` profiles.
-
-```river
-pyroscope.write "example" {
-  endpoint {
-    url = "http://pyroscope:4040"
-  }
-}
-
-pyroscope.scrape "default_settings" {
-  targets    = [ { "__address__" = "http://localhost:12345", "service_name" = "example_service" } ]
-  forward_to = [pyroscope.write.example.receiver]
-  profiling_config {
-    profile.goroutine {
-      enabled = true
-      path = "/debug/pprof/goroutine"
-      delta = false
-    }
-    profile.process_cpu {
-      enabled = true
-      path = "/debug/pprof/profile"
-      delta = true
-    }
-    profile.godeltaprof_memory {
-      enabled = true
-      path = "/debug/pprof/delta_heap"
-    }
-    profile.memory {
-      enabled = false
-      path = "/debug/pprof/heap"
-      delta = false
-    }
-    profile.godeltaprof_mutex {
-      enabled = true
-      path = "/debug/pprof/delta_mutex"
-    }
-    profile.mutex {
-      enabled = false
-      path = "/debug/pprof/mutex"
-      delta = false
-    }
-    profile.godeltaprof_block {
-      enabled = true
-      path = "/debug/pprof/delta_block"
-    }
-    profile.block {
-      enabled = false
-      path = "/debug/pprof/block"
-      delta = false
-    }
-  }
-}
-
+Or if you use gorilla/mux:
+```go
+var router *mux.Router
+router.PathPrefix("/debug/pprof").Handler(http.DefaultServeMux)
 ```
 
```

Comment:
What are the next steps for someone? is there a page that we can link them to for the next step in what they need to do? (Or maybe we need to write one?)

Reply:
not sure I understand your question. The last steps are: see profiles in browser, optimize your application, repeat

Reply:
Sometimes there is a series of tasks you need to do. Those tasks may be on multiple pages. The "next step" can be a link to the procedure the user needs to perform next. if that isn't the case here, then we don't need a link to another page. 

However, if there are things a user should repeat to refine the results, we might want to mention that.